contracts/src/token/common/erc2981.rs

```diff
@@ -0,0 +1,551 @@
+//! Implementation of the NFT Royalty Standard, a standardized way to retrieve
+//! royalty payment information.
+//!
+//! Royalty information can be specified globally for all token ids via
+//! [`Erc2981::_set_default_royalty`], and/or individually for specific token
+//! ids via [`Erc2981::_set_token_royalty`]. The latter takes precedence over
+//! the first.
+//!
+//! Royalty is specified as a fraction of sale price.
+//! [`Erc2981::fee_denominator`] is overridable but defaults to 10000, meaning
+//! the fee is specified in basis points by default.
+//!
+//! IMPORTANT: ERC-2981 only specifies a way to signal royalty information and
+//! does not enforce its payment.
+//!
+//! See <https://eips.ethereum.org/EIPS/eip-2981#optional-royalty-payments> in the ERC.
+//! Marketplaces are expected to voluntarily pay royalties together with sales,
+//! but note that this standard is not yet widely supported.
+
+use alloc::vec::Vec;
+
+use alloy_primitives::{Address, FixedBytes, U256};
+pub use sol::*;
+use stylus_sdk::{
+    prelude::{public, storage, Erase, TopLevelStorage},
+    storage::{StorageAddress, StorageMap},
+    stylus_proc::SolidityError,
+};
+
+use crate::utils::{
+    introspection::erc165::{Erc165, IErc165},
+    structs::checkpoints::{Size, S160},
+};
+
+type U96 = <S160 as Size>::Key;
+type StorageU96 = <S160 as Size>::KeyStorage;
+
+#[cfg_attr(coverage_nightly, coverage(off))]
+mod sol {
+    use alloy_sol_macro::sol;
+
+    sol! {
+        /// Indicates an error for Invalid Default Royalty. Occurs when
+        /// `numerator` > `denominator`.
```

Comment:
```suggestion
        /// Indicates that the default royalty set is invalid (eg. (numerator / denominator) >= 1).
```
update all error descriptions using the following format: `Indicates ` + the description in Solidity version (e.g. for [this error](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/3bdc3a35c504396c7227cecc32f50ae07da7e5c1/contracts/token/common/ERC2981.sol#L32C13-L32C84)).